### PR TITLE
LG-4023 Updated Acuant Logging to give more easily useful information…

### PR DESCRIPTION
**Pull request by amathews-fs**
… about the response.

### CODE REVIEW

File: lib/identity_doc_auth/acuant/responses/get_results_response.rb [hunk 1]
```diff
@@ -19,8 +19,12 @@ def initialize(http_response, config)
             extra: {
               result: result_code.name,
               billed: result_code.billed,
+              processed_alerts: processed_alerts,
+              alert_failure_count: processed_alerts[:failed].count,
```
Comment (zachmargolis) [inline lib/identity_doc_auth/acuant/responses/get_results_response.rb]: do we need to make this nil-safe?

```suggestion
              alert_failure_count: processed_alerts[:failed]&.count.to_i,
```

File: lib/identity_doc_auth/acuant/responses/get_results_response.rb [hunk 2]
```diff
@@ -75,6 +83,29 @@ def raw_alerts
...
+          parsed_response_body['Images'] || []
+        end
+
+        def processed_alerts
+          @processed_alerts ||= process_raw_alerts(raw_alerts)
+        end
+
+        def process_images_data
+          raw_images_data.index_by do |image|
+            image.except!('Uri')
```
Comment (zachmargolis) [inline lib/identity_doc_auth/acuant/responses/get_results_response.rb]: IMO if we're just mutating, I think just deleting the key would be clearer?

```suggestion
            image.delete('Uri')
```

File: lib/identity_doc_auth/acuant/responses/get_results_response.rb [hunk 3]
```diff
@@ -94,6 +125,47 @@ def attention_with_barcode?
...
+          alerts.each do |raw_alert|
+            region_refs = raw_alert['RegionReferences']
+            new_alert = {
+              name: raw_alert['Key'],
+              result: IdentityDocAuth::Acuant::ResultCodes.from_int(raw_alert['Result']).name
+            }
+
+            new_alert.merge!(get_region_info(region_refs)) if region_refs.present?
+
+            if new_alert[:result] != IdentityDocAuth::Acuant::ResultCodes::PASSED.name
```
Comment (zachmargolis) [inline lib/identity_doc_auth/acuant/responses/get_results_response.rb]: seems weird to me to parse the result code and throw it away? WDYT of:

```suggestion
            result_code = IdentityDocAuth::Acuant::ResultCodes.from_int(raw_alert['Result'])
            new_alert = {
              name: raw_alert['Key'],
              result: result_code.name
            }

            new_alert.merge!(get_region_info(region_refs)) if region_refs.present?

            if result_code != IdentityDocAuth::Acuant::ResultCodes::PASSED
```

File: lib/identity_doc_auth/acuant/responses/get_results_response.rb (hunk 1 above)
Comment (amathews-fs) [reply lib/identity_doc_auth/acuant/responses/get_results_response.rb]: I don't think so, but better safe than sorry.

File: lib/identity_doc_auth/acuant/responses/get_results_response.rb (hunk 2 above)
Comment (amathews-fs) [reply lib/identity_doc_auth/acuant/responses/get_results_response.rb]: Yeah, I'm not certain why I went with except! probably just the first thing that came into my head.

File: lib/identity_doc_auth/acuant/responses/get_results_response.rb (hunk 3 above)
Comment (amathews-fs) [reply lib/identity_doc_auth/acuant/responses/get_results_response.rb]: Yeah, thanks. That's cleaner.

File: lib/identity_doc_auth/acuant/responses/get_results_response.rb [hunk 4]
```diff
@@ -35,7 +39,11 @@ def to_h
             exception: exception,
             result: result_code.name,
             billed: result_code.billed,
+            processed_alerts: processed_alerts,
+            alert_failure_count: processed_alerts[:failed].count,
```
Comment (zachmargolis) [inline lib/identity_doc_auth/acuant/responses/get_results_response.rb]: ditto about the nil-safeness here:

```suggestion
            alert_failure_count: processed_alerts[:failed]&.count.to_i,
```

**Review by zachmargolis — APPROVED** [review]
LGTM

File: lib/identity_doc_auth/acuant/responses/get_results_response.rb (hunk 4 above)
Comment (amathews-fs) [reply lib/identity_doc_auth/acuant/responses/get_results_response.rb]: Doh. I thought about that and then forgot to make the change. Thanks.

**Pull request merged**